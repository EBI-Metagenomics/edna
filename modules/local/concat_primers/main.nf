
process CONCAT_PRIMERS {
    // Concatenate result of both the std primer identification and auto primer identification
    // This is important in case for a run the fwd strand has a known std primer but the reverse
    // strand has an auto primer
    tag "$meta.id"
    label 'extremely_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
        "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"

    input:
    tuple val(meta), path(std_primers), path(auto_primers)

    output:
    tuple val(meta), path("*concat_primers.fasta") , optional: true, emit: concat_primers_out
    path "versions.yml"                            , emit: versions

    script:
    """
    if [[ -s ${auto_primers} ]]
    then
        cat ${std_primers} > ${meta.id}_${meta.var_region}_concat_primers.fasta
        echo -en "\\n" >> ${meta.id}_${meta.var_region}_concat_primers.fasta
        cat ${auto_primers} >> ${meta.id}_${meta.var_region}_concat_primers.fasta
    else
        cat ${std_primers} > ${meta.id}_${meta.var_region}_concat_primers.fasta
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: ${params.mpt_version}
    END_VERSIONS
    """

}
