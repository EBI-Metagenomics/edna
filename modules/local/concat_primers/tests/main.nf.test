nextflow_process {

    name "Test Process CONCAT_PRIMERS"
    script "../main.nf"
    process "CONCAT_PRIMERS"
    tag "modules"
    tag "concat_primers"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false, var_region:'16S-V3-V4' ], // meta map
                    file('${projectDir}/modules/local/concat_primers/tests/data/test_std_primers.fasta', checkIfExists: true),
                    file('${projectDir}/modules/local/concat_primers/tests/data/test_auto_primers.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.concat_primers_out).match("concat_primers_out") }
            )
        }

    }

}
