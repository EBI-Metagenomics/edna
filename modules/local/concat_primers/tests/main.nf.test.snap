{
    "concat_primers_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false,
                        "var_region": "16S-V3-V4"
                    },
                    "test_16S-V3-V4_concat_primers.fasta:md5,f26cdc22a5231d2dbd0831f1f1983dc6"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T16:04:49.245847022"
    }
}